scrapyd_client/deploy.py

```diff
@@ -213,48 +214,40 @@ def _upload_egg(target, eggpath, project, version):
         'Content-Type': content_type,
         'Content-Length': str(len(body)),
     }
-    req = Request(url, body, headers)
-    _add_auth_header(req, target)
+    skip_tls_verify = target.get("skip-tls-verify", False)
+    if skip_tls_verify:
+        host_header = target.get("host")
+        if not host_header:
+            _fail("""Missing host header with skip-tls-verify active! 
+                    Please set the host in your scrapy.cfg when setting 
+                    skip-tls-verify to true!.""")
+        headers["Host"] = host_header
+    _add_auth_header(headers, target)
     _log('Deploying to project "%s" in %s' % (project, url))
-    return _http_post(req)
+    res = requests.post(
+        url, data=body, headers=headers, verify=not skip_tls_verify
+    )
+    if res.ok:
+        _log("Server response (%s):" % res.status_code)
+    else:
+        _log("Deploy failed (%s):" % res.status_code)
+    _log(json.dumps(res.json(), indent=3))
```

Comment:
This code is not equivalent to `_http_post`, which `print()`'ed successful responses without indentation, didn't print anything on `URLError` and tried to do more on `HTTPError`.

I recommend just keeping `_http_post` as-is and making this a much more minimal PR.